build.gradle

```diff
@@ -71,144 +72,174 @@ configurations {
+    }
 }
 
 tasks.withType(AbstractPublishToMaven) {
-  dependsOn checkArchiveManifests
+    dependsOn checkArchiveManifests
 }
 
 defaultTasks.add 'test'
 defaultTasks.add 'jpi'
 
 task createWrapperZip(type: Zip) {
-  archiveFileName = 'wrapper.zip'
-  destinationDirectory = new File(sourceSets.test.output.resourcesDir, 'gradle')
+    archiveFileName = 'wrapper.zip'
+    destinationDirectory = new File(sourceSets.test.output.resourcesDir, 'gradle')
 
-  from(project.rootDir) {
-    include('gradle/**')
-    include('gradlew*')
-  }
+    from(project.rootDir) {
+        include('gradle/**')
+        include('gradlew*')
+    }
 }
 
 processTestResources.dependsOn(createWrapperZip)
 
 task generateVersionFile() {
-  var versionsDir = layout.buildDirectory.dir('versions')
-  doLast {
-      def outputDir = versionsDir.get().asFile
-      outputDir.mkdirs()
+    var versionsDir = layout.buildDirectory.dir('versions')
+    doLast {
+        def outputDir = versionsDir.get().asFile
+        outputDir.mkdirs()
 
-      new File(outputDir, "gradle-enterprise-maven-extension-version.txt").text = "${gradleEnterpriseMavenExtensionVersion}"
-      new File(outputDir, "common-custom-user-data-maven-extension-version.txt").text = "${commonCustomUserDataMavenExtensionVersion}"
-  }
+        new File(outputDir, "gradle-enterprise-maven-extension-version.txt").text = "${gradleEnterpriseMavenExtensionVersion}"
+        new File(outputDir, "common-custom-user-data-maven-extension-version.txt").text = "${commonCustomUserDataMavenExtensionVersion}"
+    }
+
+    inputs.property('gradleEnterpriseMavenExtensionVersion', gradleEnterpriseMavenExtensionVersion)
+    inputs.property('commonCustomUserDataMavenExtensionVersion', commonCustomUserDataMavenExtensionVersion)
+
+    outputs.dir(versionsDir)
+        .withPropertyName('versionsDir')
+}
+
+def mavenVersions = [
+// Uncomment after 1.15.3 is released
+//    '3.1.1',
+//    '3.2.1',
+//    '3.2.5',
+    '3.3.1',
+    '3.5.4',
+    '3.8.6'
+]
+
+task downloadMavenVersions(type: Download) {
+    src(mavenVersions.collect { "https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/${it}/apache-maven-${it}-bin.zip" })
+    dest layout.buildDirectory.dir('maven')
+}
+
+task generateTestMavenVersions() {
+    var versionsDir = layout.buildDirectory.dir('maven-test-versions')
+    doLast {
+        def outputDir = versionsDir.get().asFile
+        outputDir.mkdirs()
 
-  inputs.property('gradleEnterpriseMavenExtensionVersion', gradleEnterpriseMavenExtensionVersion)
-  inputs.property('commonCustomUserDataMavenExtensionVersion', commonCustomUserDataMavenExtensionVersion)
+        new File(outputDir, "maven-test-versions.txt").text = "${mavenVersions.join(',')}"
+    }
 
-  outputs.dir(versionsDir)
-      .withPropertyName('versionsDir')
+    inputs.property('mavenVersions', mavenVersions)
+
+    outputs.dir(versionsDir)
+        .withPropertyName('versionsDir')
```

Comment:
How this property is used?

Reply:
It's a good ideea to declare inputs and outputs for tasks so that incremental build works properly. However this task is not needed anymore since removed the download.